internal/driver/nodeserver.go

```diff
@@ -380,16 +382,21 @@ func (ns *NodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 	// from the limit of block devices that can be attached to the instance,
 	// which will effectively give us the number of block storage volumes
 	// that can be attached to this node/instance.
-	//
-	// This is what the spec wants us to report: the actual number of volumes
-	// that can be attached, and not the theoretical maximum number of
-	// devices that can be attached.
 	log.V(4).Info("Listing instance disks", "nodeID", ns.metadata.ID)
 	disks, err := ns.client.ListInstanceDisks(ctx, ns.metadata.ID, nil)
 	if err != nil {
 		return &csi.NodeGetInfoResponse{}, errInternal("list instance disks: %v", err)
 	}
-	maxVolumes := maxVolumeAttachments(ns.metadata.Memory) - len(disks)
+
+	// Filter out disks that are created by the driver
+	// to avoid counting them towards the max volume attachments
+	// that can be attached to the instance.
+	filterdDisks := slices.DeleteFunc(disks, func(disk linodego.InstanceDisk) bool {
+		// Check if the disk label starts with the volume label prefix
+		return strings.HasPrefix(disk.Label, ns.driver.volumeLabelPrefix)
+	})
```

Comment:
I'm not sure if we want to filter out the volume that are attached to the node here. This would result in incorrect number of volumes that can be further attached. No?

Reply:
[The CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#nodegetinfo) is unfortunately not as explicit as it could be. However, it seems intended that you return the maximum # of volumes that can be attached (irrespective of how many volumes are _currently_ attached).

[Example](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/a25f651b6f2e39ba1ef79dbac4d638f36ad62292/pkg/driver/node.go#L767-L810) of the AWS EBS CSI driver calculating the volume limit

Reply:
You can see [here](https://github.com/kubernetes/kubernetes/blob/bf70d289fb3b7deb9b121bbe10e278f12840cfcb/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L204-L258) that the kube-scheduler calculates how many volumes are currently attached and compares that to the limit (which we return here) of how many total volumes _can_ be attached.

Reply:
Ah good catch. Thanks for sharing that. Yeah looking into it further I agree with you. We should make sure that we do not count for attached volumes (only the disks - boot and swap or just boot in LKE). 

But I believe we are doing that correctly already. ListInstanceDisk() only return the disks and do not include the volumes attached to the node. So we don't really need to filter for it.